modules/library/metadata/src/main/java/org/geotools/factory/FactoryCreator.java

```diff
@@ -21,6 +21,8 @@
 import java.lang.ref.WeakReference;
 import java.lang.reflect.Modifier;
 import java.lang.reflect.InvocationTargetException;
+import java.util.function.Predicate;
+import javax.imageio.spi.ServiceRegistry.Filter;
```

Comment:
We may need to make our own copy of this interface, so that this infrastructure can be used on systems where imageio is not available.\

(this is a blocker for people trying to use GeoTools on Andriod for example)

Reply:
Seems like a good time to do that. But what exactly do you mean by "our own copy"? Does that include the namespace/package or just the interface's functionality?